src/3d/symbols/qgsline3dsymbol_p.cpp

```diff
@@ -146,87 +148,84 @@ void QgsBufferedLine3DSymbolHandler::processFeature( const QgsFeature &f, const
     width = 0.001;
   }
 
-  QgsAbstractGeometry *buffered = engine.buffer( width / 2., nSegments, endCapStyle, joinStyle, mitreLimit ); // factory
+  std::unique_ptr<QgsAbstractGeometry> buffered( engine.buffer( width / 2., nSegments, endCapStyle, joinStyle, mitreLimit ) ); // factory
   if ( !buffered )
     return;
 
   if ( QgsWkbTypes::flatType( buffered->wkbType() ) == Qgis::WkbType::Polygon )
   {
-    QgsPolygon *polyBuffered = static_cast<QgsPolygon *>( buffered );
-    processPolygon( polyBuffered, f.id(), mSymbol->offset(), mSymbol->extrusionHeight(), context, out );
+    std::unique_ptr<QgsPolygon> polyBuffered( qgsgeometry_cast<QgsPolygon *>( buffered.get() )->clone() );
```

Comment:
This is a speed regression -- it was intentionally written the original way to avoid an unnecessary clone

Reply:
Removed the commit.